src/GslCore/PragmaTypes.fs

```diff
@@ -102,144 +102,284 @@ let parsePlatform (args: string list) =
 
 let warningPragmaDef =
-    {name = "warn"; argShape = AtLeast(1); scope = BlockOrPart(Transient);
-     desc = "Print a warning message.";
-     invertsTo = None; validate = noValidate}
+    { name = "warn"
+      argShape = AtLeast(1)
+      scope = BlockOrPart(Transient)
+      desc = "Print a warning message."
+      invertsTo = None
+      validate = noValidate }
 
 let warnoffPragmaDef =
-    {name = "warnoff"; argShape = One; scope = BlockOnly(Persistent);
-     desc = "Turn off specific warnings.";
-     invertsTo = None; validate = noValidate} // TODO parse function here
+    { name = "warnoff"
+      argShape = One
+      scope = BlockOnly(Persistent)
+      desc = "Turn off specific warnings."
+      invertsTo = None
+      validate = noValidate } // TODO parse function here
 
 let capaPragmaDef =
-    {name = "capa"; argShape = One; scope = BlockOnly(Persistent);
-     desc = "Enables particular compiler capabilities.";
-     invertsTo = None; validate = noValidate}
+    { name = "capa"
+      argShape = One
+      scope = BlockOnly(Persistent)
+      desc = "Enables particular compiler capabilities."
+      invertsTo = None
+      validate = noValidate }
 
 let platformPragmaDef =
-    {name = "platform"; argShape = One; scope = BlockOnly(Persistent);
-     desc = "Specify an assembly platform to target, current options: 'stitch', 'megastitch'.";
-     invertsTo = None; validate = parsePlatform >> (lift ignore)}
+    { name = "platform"
+      argShape = One
+      scope = BlockOnly(Persistent)
+      desc = "Specify an assembly platform to target, current options: 'stitch', 'megastitch'."
+      invertsTo = None
+      validate = parsePlatform >> (lift ignore) }
 
 let markersetPragmaDef =
-    {name = "markerset"; argShape = One; scope = BlockOnly(Persistent);
-     desc = "Set the default marker set for a ### part.";
-     invertsTo = None; validate = noValidate}
+    { name = "markerset"
+      argShape = One
+      scope = BlockOnly(Persistent)
+      desc = "Set the default marker set for a ### part."
+      invertsTo = None
+      validate = noValidate }
 
 let namePragmaDef =
-    {name = "name"; argShape = One; scope = BlockOrPart(Transient);
-     desc = "Override name for a assembly or part.";
-     invertsTo = None; validate = noValidate}
+    { name = "name"
+      argShape = One
+      scope = BlockOrPart(Transient)
+      desc = "Override name for a assembly or part."
+      invertsTo = None
+      validate = noValidate }
 
 let fusePragmaDef =
-    {name = "fuse"; argShape = Zero; scope = PartOnly;
-     desc = "Create a seamless junction with the next part.";
-     invertsTo = None; validate = noValidate}
+    { name = "fuse"
+      argShape = Zero
+      scope = PartOnly
+      desc = "Create a seamless junction with the next part."
+      invertsTo = None
+      validate = noValidate }
+
+let topologyPragmaDef =
+    { name = Topology.PragmaName
+      argShape = One
+      scope = BlockOnly(Persistent)

```

Comment:
I'm surprised that this is persistent; seems like this should be block-transient, otherwise if a function expands into an assembly with a different topology it will impact all of the assemblies that appear after it in the parse tree.

Reply:
@daz10000 is the idea here that we expect an entire compilation to produce either circular or linear constructs?  Per-assembly annotation makes more sense to me.

Reply:
> I'm surprised that this is persistent; seems like this should be block-transient, otherwise if a function expands into an assembly with a different topology it will impact all of the assemblies that appear after it in the parse tree.

I thought functions isolated pragma scope already?  This would be a problem for all persistent pragmas like platform.   I do think from a user point of view that they are likely to continue for a while with the same topology .e.g a whole bunch of plasmids, in the same way #platform stitch / megastitch would be used.

Reply:
> @daz10000 is the idea here that we expect an entire compilation to produce either circular or linear constructs? Per-assembly annotation makes more sense to me.

No - it's assembly level information, but I think ( you should ask some folks and do some market research), likely to be done in blocks.  Typical use cases

1.  I have a whole bunch of plasmids
#toplogy circular
p1 assembly
p2 assembly 
p3 assembly

// also some linear dna
#topology linear
l1 assembly
l2 assembly

Guides are a good use case.  If there's a function for emitting a guide RNA, it would tag the assembly circular maybe but you wouldn't want that to escape the function scope.

Reply:
You're correct about function scope, I thought I went back and commented as such but looks like it got lost.  We can roll with it for now as persistent.